src/chart/pie/PieView.ts

```diff
@@ -193,7 +193,7 @@ class PiePiece extends graphic.Sector {
 
         const labelPosition = seriesModel.get(['label', 'position']);
         if (labelPosition !== 'outside' && labelPosition !== 'outer') {
-            sector.getTextGuideLine()?.hide();
+            sector.removeTextGuideLine();
             return;
         }
```

Comment:
Seems the guideline can't be restored if the position is changed from 'inside' to 'outside' again. Perhaps we need to recreate the guideline here instead of in https://github.com/apache/echarts/blob/2c88dec314fb8260c6957341df174da8cdb9daea/src/chart/pie/PieView.ts#L49
```ts
if (..)  { ... }
else {
  let polyline = this.getTextGuideLine();
  if (!polyline) {
    polyline = new graphic.Polyline();
    this.setTextGuideLine(polyline);
  }
  // Default use item visual color
   setLabelLineStyle(this, getLabelLineStatesModels(itemModel), {
       stroke: visualColor,
       opacity: retrieve3(labelLineModel.get(['lineStyle', 'opacity']), visualOpacity, 1)
    });
}
```

Reply:
Thanks for the review @pissang ! This change makes sense, and I did the proposed changes. I also added yet another flip-flop (inside/outside) to the test to make sure it can change back yet one more time, just in case.

Reply:
The relevant updated test:
![pie-after2](https://user-images.githubusercontent.com/33317356/115338749-26109580-a1ac-11eb-83bf-f0ed2b9bd40a.gif)